controller_adapters/database.go

```diff
@@ -82,3 +84,102 @@ func (d *Database) GetProfileResolvers() ndb_api.ProfileResolvers {
 	return profileResolvers
 
 }
+
+// Returns basic details about the Time Machine if provided in the
+// underlying database, else returns defaults like:
+// TM Name: db_instance_name_TM
+// TM Description: Time Machine for db_instance_name
+// SLA Name: NONE
+func (d *Database) GetTMDetails() (tmName, tmDescription, slaName string) {
+	tmInfo := d.Spec.Instance.TMInfo
+
+	tmName = tmInfo.Name
+	tmDescription = tmInfo.Description
+	slaName = tmInfo.SLAName
+
+	if tmName == "" {
+		tmName = d.GetDBInstanceName() + "_TM"
+	}
+	if tmDescription == "" {
+		tmDescription = "Time Machine for " + d.GetDBInstanceName()
+	}
+	if slaName == "" {
+		slaName = common.SLA_NAME_NONE
+	}
+
+	return
+}
+
+// Returns a schedule struct for the time machine.
+func (d *Database) GetTMSchedule() (schedule ndb_api.Schedule, err error) {
+	tmInfo := d.Spec.Instance.TMInfo
+	hhmmssDaily := strings.Split(tmInfo.DailySnapshotTime, ":")
```

Comment:
can we factor out the ability to parse `tmInfo.DailySnapshotTime` in hh, mm, ss into a separate fn?

Reply:
Sure, will move that out to a non-exported function.

Reply:
I realized that there's a `time.Parse` function available in golang, replaced the code block with that instead.

`hhmmss, err := time.Parse(time.TimeOnly, tmInfo.DailySnapshotTime)
	if err != nil {
		err = errors.Join(err, errors.New("error converting daily snapshot time"))
		return
	}
	hh, mm, ss := hhmmss.Hour(), hhmmss.Minute(), hhmmss.Second()`